libs/blocks/merch/merch.js

```diff
@@ -116,17 +124,62 @@ export async function getPriceContext(el, params) {
   };
 }
 
+let loadEntitlementsPromise;
+
+export async function loadEntitlements() {
+  await loadIms();
+  if (!window.adobeIMS?.isSignedInUser?.()) return Promise.resolve([{}, {}]);
+  const [{ default: getUserEntitlements },
+    mappings] = await Promise.all([
+    import('../global-navigation/utilities/getUserEntitlements.js'),
+    fetch(entitlementsURL),
+  ]);
+  const entitlements = await getUserEntitlements();
+  const entitlementsMetadata = await mappings.json();
+  return [entitlements, entitlementsMetadata];
+}
+
+export async function handleEntitlements(cta, promise) {
+  const [{ arrangement_codes: aCodes }, entitlementsMappings] = await promise;
+  if (aCodes === undefined) return;
+  await cta.onceSettled();
+  const { value: [{ productArrangementCode }] } = cta;
+  if (aCodes[productArrangementCode] === true) {
+    const mapping = entitlementsMappings.data
+      ?.find(({ [TITLE_PRODUCT_ARRANGEMENT_CODE]: code }) => code === productArrangementCode);
+    if (!mapping) return;
+    cta.firstElementChild.innerHTML = (await replaceKey(mapping.CTA, getConfig()))
+         || cta.firstElementChild.innerHTML;
+    cta.href = mapping.Target;
```

Comment:
Any chance `mapping.Target` might be `undefined`?

Reply:
This is an excel file owned by GWP,  so it should not be.
This is for logged-in only use case.
Still I'll keep the original value as fallback.